lib/execution_engine2/SDKMethodRunner.py

```diff
@@ -838,9 +838,12 @@ def check_jobs(self, job_ids, check_permission=True, projection=None):
 
             job_states[str(job.id)] = mongo_rec
 
+        if return_list is not None and SDKMethodRunner.parse_bool_from_string(return_list):
+            job_states = list(job_states.values())
```

Comment:
I think you need to iterate over the job ids, like 100,900,200,9999 and return them in that order like
list = [{job_state_for_id_100}, {job_state_for_id_900}, {'maybe this job state is missing'},{job_state_for_id_999}]

Reply:
@bio-boris I added a `order_by` in `get_jobs`. So the default jobs returned are sorted by id.